### PR TITLE
Add kubernetes-csi-external-provisioner image

### PR DESCRIPTION
**Pull request by rawlingsj**
- [x] **IMPORTANT: 'image-request' tag has been applied if this PR is adding any images, including new versions or variants**

#### For new image PRs only

If you have an apko.yaml file in this PR you need to follow this checklist, otherwise feel free to remove.
- [x] Include tests, sufficient enough that you would trust this image running in production.
- [x] The version included is the latest GA version of the software
- [x] The latest tag points to the newest stable version
- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] The image runs as `nonroot` and GID/UID are set to 65532
  - [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [x] ENTRYPOINT
  - [ ] For applications/servers/utilities call main program with no arguments e.g. [redis-server]
  - [ ] For base images leave empty
  - [ ] For dev variants set to entrypoint script that falls back to system
- [ ] CMD:
  - [ ] For server applications give arguments to start in daemon mode (may be empty)
  - [ ] For utilities/tooling bring up help e.g. `–help`
  - [ ] For base images with a shell, call it e.g. [/bin/sh]
- [ ] Consider where and how the image deviates from popular alternatives. Is there a good reason and is it documented?
- [x] Add annotations e.g:
- [x] Check if environment variables are needed e.g. to set data locations
- [ ] Ensure the image responds to SIGTERM
  - [ ] `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`
- [x] Documentation. Let's make this excellent. Include usage example.
- [x] Error logs write to stderr and normal logs to stdout. DO NOT write to file.

### CODE REVIEW

**Comment by rawlingsj** [comment]
Opening in draft to get eyes on this as it's been a while since I created an image in this repo.  

Tests will fail as it is dependant on https://github.com/wolfi-dev/os/pull/2060

File: images/kubernetes-csi-external-provisioner/configs/latest.apko.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+accounts:
+  groups:
+    - groupname: nonroot
+      gid: 65532
+  users:
+    - username: nonroot
+      uid: 65532
+      gid: 65532
+  run-as: root # image is not working with nonroot user
```
Comment (rawlingsj) [inline images/kubernetes-csi-external-provisioner/configs/latest.apko.yaml:18]: This is something to be aware of, I've not yet been able to run as `nonroot`

Comment (joshrwolf) [reply images/kubernetes-csi-external-provisioner/configs/latest.apko.yaml:18]: I think this is ok for this, and the other csi related things that need to mount to a host socket

File: images/kubernetes-csi-external-provisioner/configs/latest.apko.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  keyring:
+    - https://packages.wolfi.dev/os/wolfi-signing.rsa.pub
+  repositories:
+    - https://packages.wolfi.dev/os
+  packages:
+    - kubernetes-csi-external-provisioner
+    - wolfi-baselayout
+
+accounts:
+  groups:
```
Comment (joshrwolf) [inline images/kubernetes-csi-external-provisioner/configs/latest.apko.yaml:11]: are `groups/users` needed at all if we're just going to use `root`?

Comment (rawlingsj) [reply images/kubernetes-csi-external-provisioner/configs/latest.apko.yaml:11]: yeah I wasn't sure about that either, @imjasonh WDYT?

File: images/kubernetes-csi-external-provisioner/configs/latest.apko.yaml (hunk 1 above)
Comment (rawlingsj) [reply images/kubernetes-csi-external-provisioner/configs/latest.apko.yaml:18]: thanks

File: images/kubernetes-csi-external-provisioner/tests/02-deploy.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+function cleanup() {
+    # Print debug logs and status
+    kubectl get pods
+    kubectl describe pods
+    
+    # Seeing intermittent failures if we don't wait for a bit here
+    # The `rollout status`` below should wait for terminated pods to be removed
+    # However, we still occasionally see a terminating pod which fails when checking logs
+    sleep 10
```
Comment (rawlingsj) [inline images/kubernetes-csi-external-provisioner/tests/02-deploy.sh:34]: This is annoying but it's been the only way I've seen the test consistently work.  The `kubectl rollout status` in the script should wait for old pods to be terminated but there's occasionally one that sticks around for a little while.  Other suggestions welcome else another sleep is the best I have ATM

**Comment by rawlingsj** [comment]
Passing kind tests https://github.com/chainguard-images/images/actions/runs/5006713567/jobs/8972408981?pr=612

**Review by joshrwolf — APPROVED** [review]

**Pull request merged**